packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -1439,7 +1454,18 @@ export class DATE extends AbstractDataType<AcceptedDate> {
       return value;
     }
 
-    if (typeof value === 'string' || typeof value === 'number') {
+    if (typeof value === 'string') {
+      if (this.options.plain) {
+        // Plain date times are stored as UTC therefore we need to strip any timezone data from the string
```

Comment:
I'm not convinced by this one. I think when the datetime is `plain`, it should forbid text inputs that specify a timezone offset. It should be in the ISO8601 format but with no timezone offset

Reply:
I'm a bit stuck on this.
Should `Z` be allowed or not?

When a string is passed to `sanitize` should the resulting `Date` object be UTC or in the server's local time?
For example what should the output of `sanitize('2023-03-04T12:00:00.123')` be:

- A Date object with an ISOString value of `2023-03-04T12:00:00.123Z` - set the value to UTC time
- A Date object with an ISOString value of `2023-03-03T23:00:00.123Z` - dependant on server time zone
- A Date object using the offset from the sequelize instances `options.timezone` value, defaulting to UTC.

Reply:
I've updated the logic now. It seems to work locally will wait for tests to confirm.

Reply:
The way I got this to work was to validate the date string in the `parseDatabaseValue` function and confirm it was ISO8601 (or similar to support `YYYY-MM-DDTHH:mm:ss.SSS` or `YYYY-MM-DD HH:mm:ss.SSS` with optional timezone) and convert it to a date object with the timezone in the string or use UTC. Then in the `sanitize` function, the same validation occurs however if the its for a plain date, and a timezone exists, an error is thrown.

Reply:
I've done some more testing and updated the functions now.
The plain date will only allow plain dates without any time zone information (i.e. Z or +00:00).